server/src/gameserver/manager.rs

```diff
@@ -9,50 +14,67 @@ pub enum Error {
     ReadinessIssue(String),
     #[error("failed to cleanly shutdown the server: `{0}`")]
     ShutdownFailure(String),
+    #[error("failed reserve the board server: `{0}`")]
+    ReservationFailed(String),
 }
 
 #[cfg(feature = "agones_sdk")]
 pub struct Manager {
-    sdk: agones::Sdk,
-    health_checker: HealthChecker,
+    sdk: Sdk,
+    _health_checker: HealthChecker,
+    spaces_sender: mpsc::Sender<usize>
 }
 
 #[cfg(feature = "agones_sdk")]
 impl Manager {
     pub async fn setup() -> Result<Manager, Error> {
         log::info!("connecting to agones sdk sidecar...");
-        let sdk = agones::Sdk::new(None, None)
+        let sdk = Sdk::new(None, None)
             .await
             .map_err(|e| Error::SetupFailure(format!("{}", e)))?;
 
-        let health_checker = HealthChecker::new(sdk.clone());
+        let health_checker = HealthChecker::new(&sdk);
+        let sender = Manager::new_spaces_channel(&sdk);
         Ok(Manager {
             sdk,
-            health_checker,
+            _health_checker: health_checker,
+            spaces_sender: sender
         })
     }
 
-    pub fn start_health_check(&mut self) {
-        log::info!("starting health check via agones...");
-        self.health_checker.start()
+    fn new_spaces_channel(sdk: &Sdk) -> mpsc::Sender<usize> {
+        let (tx, mut rx) = mpsc::channel::<usize>(10);
+        let mut my_sdk = sdk.clone();
+        task::spawn(async move {
+            loop {
+                while let Some(space_id) = rx.recv().await {
+                    log::debug!("Reserving for space ID: {:?}", space_id);
+                    my_sdk.reserve(Duration::from_secs(30))
+                        .await
+                        .expect(format!("Can't reserve space {}", space_id).as_str())
+                }
+            }
+        });
+
+        tx
+    }
+
+    pub fn spaces_handle(&self) -> mpsc::Sender<usize> {
```

Comment:
focusing on agones_sdk impl. now, will retrofit the `not(agones_sdk)` when the sdk integration is there